pages/edge.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import Page from '../components/edge/Page';
+import Download from '../components/edge/Download';
+import Link from '../components/edge/Link';
+import Gain from '../components/edge/Gain';
+import { withTranslation } from '../i18n';
+import { string } from 'prop-types';
+
+class Edge extends React.Component<any> {
+  static async getInitialProps() {
+    return {
+      namespacesRequired: ['common'],
+    };
+  }
+
+  constructor(props) {
```

Comment:
The component only re-renders the views when the props or state changes so you might use state for the "sn".

```ts
class Edge extends React.Component<any> {
   state = {
     sn: '',
  }

  async componentDidMount() {
     ...
      this.setState({
        sn,
     })
  }
```